low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server_manifest.py

```diff
@@ -0,0 +1,335 @@
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com
+
+import errno
+import time
+import json
+import os
+
+from cortx.utils.process import SimpleProcess
+from cortx.utils.kv_store import KvStoreFactory
+from framework.utils.conf_utils import GLOBAL_CONF, Conf, NODE_TYPE_KEY
+from error import ManifestError
+from resource_map import CortxManifest
+from framework.base.sspl_constants import (MANIFEST_SVC_NAME, LSHW_FILE,
+    MANIFEST_OUTPUT_FILE)
+from framework.utils.service_logging import CustomLog, logger
+from framework.utils.utility import Utility
+from framework.platforms.server.software import Service
+from framework.platforms.server.platform import Platform
+
+
+class ServerManifest(CortxManifest):
+    """ServerManifest class provides resource map and related information
+    like health.
+    """
+
+    name = "server_manifest"
+
+    def __init__(self):
+        """Initialize server manifest."""
+        super().__init__()
+        self.log = CustomLog(MANIFEST_SVC_NAME)
+        server_type = Conf.get(GLOBAL_CONF, NODE_TYPE_KEY)
+        Utility.validate_server_type_support(self.log, ManifestError, server_type)
+        self.field_mapping = {
+            'id': 'uid',
+            'class': 'type',
+            'description': 'description',
+            'product': 'product',
+            'serial': 'serial_number',
+            'vendor': 'manufacturer',
+            'part_number': 'part_number',
+            'model_number': 'model_number',
+            'physid': 'physid',
+            'version': 'version'
+        }
+        self.class_mapping = {
+            'memory': 'hw>memory[%s]>%s',
+            'disk': 'hw>disk[%s]>%s',
+            'storage': 'hw>storage[%s]>%s',
+            'system': 'hw>system[%s]>%s',
+            'processor': 'hw>processor[%s]>%s',
+            'network': 'hw>network[%s]>%s',
+            'power': 'hw>power[%s]>%s',
+            'volume': 'hw>volumes[%s]>%s',
+            'bus': 'hw>bus[%s]>%s',
+            'bridge': 'hw>bridge[%s]>%s',
+            'display': 'hw>display[%s]>%s',
+            'generic': 'hw>generic[%s]>%s',
+            'input': 'hw>input[%s]>%s'
+        }
+        self.kv_dict = {}
+        sw_resources = {
```

Comment:
We need to avoid working in hierarchy and work strictly out of the mapping table.

Reply:
Yes, we strictly working with the mapping tables only.
just to maintain resource types in logical code purpose, we have defined the above filelds.